enhancements/kube-apiserver/bound-sa-tokens.md

```diff
@@ -0,0 +1,153 @@
+        audience will be defaulted to the issuer.
+  - Enabling bound tokens will satisfy goals #1 and #2.
+
+- The apiserver operator should manage a keypair used to sign and verify bound tokens
+  - The keypair needs to be distinct from the keypair used for legacy service account
+    tokens. Reusing the legacy token keypair would prevent token validation since tokens
+    are first validated by key and then by content. If the key were to validate, the
+    subsequent content check would fail due to differences in the content of legacy and
+    bound tokens.
+  - The keypair should be written to a secret in the `openshift-kube-apiserver` namespace
+  - The public key should be added to (rather than replaced in) a configmap in the
+    `openshift-config-managed` namespace.
+    - This configmap can be used to source the public keys needed by 3rd party components
+      to verify bound tokens issued by the apiserver, satisfying goal #3.
+    - The public keys in the configmap should be included in
+      `KubeAPIServerConfig.ServiceAccountPublicKeyFiles` for provision to the apiserver.
+      - In the event that the keypair secret is deleted, a new keypair will be generated
+        but tokens signed by the previous private key will continue to validate since the
+        corresponding public key will still be used to validate bound tokens.
+    - (Optional) Public keys known to be older than the maximum expiry of previously
+      issued tokens could be removed from the configmap.
+
+### Implementation Details/Notes/Constraints [optional]
+
+Bound tokens are not currently required until bootstrap is complete, so it should be
+reasonable to delay support of bound tokens until the post-bootstrap phase.
+
+### Risks and Mitigations
+
+Configuring a cluster to support `TokenRequest` and `TokenRequestProjection` should have
+no impact on the usage of legacy tokens. Existing service account tokens will continue to
+work as before, and bound tokens will be supported for those that explicitly request
+them. Bound tokens only replace legacy tokens if the `BoundServiceAccountTokenVolume`
+feature gate is enabled. and it is disabled by default. Existing tests passing should be
+a good indication that legacy token usage is unaffected.
+
+## Design Details
+
+### Test Plan
+
+There is already comprehensive test coverage of the bound token feature in upstream
+kube. It should therefore be reasonable to limit test coverage to indications that bound
+token usage has been properly configured:
+ - Validating that a bound token can be requested via the TokenRequest API
+ - Validating that a pod can request a bound token via volume projection
+
+It probably makes sense to manually verify integration with the AWS pod identity webhook,
+since this is the primary motivation for this enhancement. It may make sense to delegate
+this testing to those that will be deploying the webhook for customers.
```

Comment:
Maybe deletion of the secret would make sense to test.

Reply:
Updated to include testing of both secret and configmap removal.